MyCryptoHQ/MyCrypto
# Add Monero swap support

Closes #1410

### Description

Adds functionality to swap between XMR -> ??? and ??? -> XMR via ShapeShift.io.

### Changes

* Adds a new abstracted UI component, `<Warning />`, to standardize warning notifications.
* Allows selection of XMR in the "deposit" and "withdraw" dropdowns on Swap.
* Shows an input field for the transaction's payment ID when it's an XMR -> ??? transaction.

### Steps to Test

1. Visit the "Swap" page.
2. Select "XMR" from either the "deposit" or "withdraw" dropdowns.
3. Perform a swap as usual.
4. If the swap is XMR -> ???, ensure the payment ID field and warning appear.

### Screenshots

<img width="1284" alt="screen shot 2018-06-25 at 5 50 29 pm" src="https://user-images.githubusercontent.com/10479826/41879818-d770bd2e-78a0-11e8-9511-457c10d877b3.png">
